core/load-git.c

```diff
@@ -725,28 +750,41 @@ static int get_divemode(const char *divemodestring) {
 	return 0;
 }
 
-static void parse_event_keyvalue(void *_event, const char *key, const char *value)
+/*
+ * A 'struct event' has a variable-sized name allocation at the
+ * end. So when we parse the event data, we can't fill in the
+ * event directly, because we don't know how to allocate one
+ * before we have the size of the name.
+ *
+ * Thus this initial 'parse_event' with a separate name pointer.
+ */
+struct parse_event {
+	const char *name;
+	struct event ev;
+};
+
+static void parse_event_keyvalue(void *_parse, const char *key, const char *value)
 {
-	struct event *event = _event;
+	struct parse_event *parse = _parse;
 	int val = atoi(value);
 
 	if (!strcmp(key, "type")) {
-		event->type = val;
+		parse->ev.type = val;
 	} else if (!strcmp(key, "flags")) {
-		event->flags = val;
+		parse->ev.flags = val;
 	} else if (!strcmp(key, "value")) {
-		event->value = val;
+		parse->ev.value = val;
 	} else if (!strcmp(key, "name")) {
-		/* We get the name from the string handling */
+		parse->name = value;
 	} else if (!strcmp(key,"divemode")) {
-		event->value = get_divemode(value);
+		parse->ev.value = get_divemode(value);
 	} else if (!strcmp(key, "cylinder")) {
 		/* NOTE! We add one here as a marker that "yes, we got a cylinder index" */
-		event->gas.index = 1 + get_index(value);
+		parse->ev.gas.index = 1 + get_index(value);
 	} else if (!strcmp(key, "o2")) {
-		event->gas.mix.o2 = get_fraction(value);
+		parse->ev.gas.mix.o2 = get_fraction(value);
 	} else if (!strcmp(key, "he")) {
-		event->gas.mix.he = get_fraction(value);
+		parse->ev.gas.mix.he = get_fraction(value);
 	} else
 		report_error("Unexpected event key/value pair (%s/%s)", key, value);
```

Comment:
not your code, but a coding style violation... should have `{ }` around the `report_error()`